gnowledge/gstudio
# Added script to create theme-topic hierarchy

**Added script to create theme-topic hierarchy:**
- Need to re-write the script again to accommodate recent required changes.
- Now the script will process CSV with CR/XCR field.
- Also previously, `content_org` filed was not there for topics but current CSV is going to have that field too.
- Script is able to process multiple levels of theme-items.
  - i.e: Schema can be varying for different rows.
  - With one constrain of last cell should be of `topic`
- Script now checks for the empty field after topic cell and discards it.
- Now `Theme` instances can have `curricular` as one of the attribute. For that changed `factory_type.py` file.

--

**TEST:**
- run: `python manage.py filldb`
- then run: `python manage.py create_theme_topic_hierarchy <file_name.csv>` 
- try to see these cards appearing under `Repository`
